### PR TITLE
fix white space

### PR DESCRIPTION
**Pull request by da1nerd**
See https://github.com/unfoldingword/translationcore/issues/7164

\p should not have any trailing spaces before the newline

I added a fix for https://github.com/unfoldingword/translationcore/issues/7165 as well since it was closely related. This removes white space after the pipe |.

There are a ton of file changes, but those are mostly updated test files.

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword/usfm-js/101)

### CODE REVIEW

**Comment by RobH123** [comment]
@da1nerd I noticed these warnings in a USFM check today, and wondered if they might also be covered by this set of changes:
LT Unexpected USFM field with 'line marker='\w'' at EXO 6:14 in en_ult repository in file 02-EXO.usfm on line 3,895 around ◗fathers:␣\p◖ (Priority 875)
LT Unexpected USFM field with 'line marker='\zaln-s'' at EXO 6:23 in en_ult repository in file 02-EXO.usfm on line 4,114 around ◗Ithamar.\p␣␣◖ (Priority 875)
LT Unexpected USFM field with 'line marker='\w'' at EXO 6:29 in en_ult repository in file 02-EXO.usfm on line 4,253 around ◗you.”\p␣␣◖ (Priority 875)
ST Unexpected USFM field with 'line marker='\w'' at EXO 6:14 in en_ust repository in file 02-EXO.usfm on line 4,120 around ◗leaders.\p◖ (Priority 875)
ST Unexpected USFM field with 'line marker='\w'' at EXO 6:14 in en_ust repository in file 02-EXO.usfm on line 4,140 around ◗them.\p␣␣◖ (Priority 875)
ST Unexpected USFM field with 'line marker='\w'' at EXO 6:29 in en_ust repository in file 02-EXO.usfm on line 4,429 around ◗you.”\p␣␣◖ (Priority 875)

Could it also be USFM-JS sometimes appending \p markers to the ends of lines and often with superfluous spaces before or after?

**Comment by da1nerd** [comment]
@RobH123 I'm making some additional tweaks to fix those bugs you mentioned.
In doing so I ended up changing some other things I'd like to run by you.

* Is it ok to eliminate all blank lines from usfm except for a blank line at the end of the file? This would include removing blank lines before \s5 markers
* Is it ok to eliminate all white space at the end of each line of usfm?

**Comment by RobH123** [comment]
Great! I'm glad I noticed those while your mind was still in here. I'd forgotten that it often did that.

No to #1. Please leave blank lines in. The editors use them for visual breaks. They do no harm.

If you wanted, I think it would be ok to normalise two or more blank lines just to one.

Please ensure that the file ends with a newline.

Yes to #2. I know of no need to ever have trailing spaces.

Thanks @da1nerd .

File: __tests__/resources/acts-1-20.aligned.oldformat.usfm [hunk 1]
```diff
@@ -1,86 +1,86 @@
 \id ACT EN_ULT en_English_ltr Wed Oct 10 2018 11:55:23 GMT-0400 (EDT) tc
 \c 1
 \v 20 "
-\zaln-s | x-strong="G10630" x-lemma="γάρ" x-morph="Gr,CC,,,,,,,," x-occurrence="1" x-occurrences="1" x-content="γὰρ"
+\zaln-s |x-strong="G10630" x-lemma="γάρ" x-morph="Gr,CC,,,,,,,," x-occurrence="1" x-occurrences="1" x-content="γὰρ"
```
Comment (PhotoNomad0) [inline __tests__/resources/acts-1-20.aligned.oldformat.usfm:4]: We should leave the spaces in this file.  This is to verify that we can still import the old file format.

File: __tests__/resources/acts_1_11.aligned.oldformat.usfm [hunk 2]
```diff
@@ -1,106 +1,106 @@
 \id ACT EN_ULT en_English_ltr Wed Oct 10 2018 11:55:23 GMT-0400 (EDT) tc
 \c 1
 \v 1
-\zaln-s | x-strong="G25320" x-lemma="καί" x-morph="Gr,D,,,,,,,,," x-occurrence="1" x-occurrences="1" x-content="καὶ"
+\zaln-s |x-strong="G25320" x-lemma="καί" x-morph="Gr,D,,,,,,,,," x-occurrence="1" x-occurrences="1" x-content="καὶ"
```
Comment (PhotoNomad0) [inline __tests__/resources/acts_1_11.aligned.oldformat.usfm:4]: ditto

File: __tests__/resources/acts_1_4.aligned.oldformat.usfm [hunk 3]
```diff
@@ -1,69 +1,69 @@
 \id ACT EN_ULT en_English_ltr Wed Oct 10 2018 11:55:23 GMT-0400 (EDT) tc
 \c 1
 \v 4
-\zaln-s | x-lemma="καί" x-morph="Gr,CC,,,,,,,," x-occurrence="1" x-occurrences="1" x-strong="G25320" x-content="καὶ"
+\zaln-s |x-lemma="καί" x-morph="Gr,CC,,,,,,,," x-occurrence="1" x-occurrences="1" x-strong="G25320" x-content="καὶ"
```
Comment (PhotoNomad0) [inline __tests__/resources/acts_1_4.aligned.oldformat.usfm:4]: ditto

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]

**Comment by PhotoNomad0** [comment]
@da1nerd This is probably ready to go with just the above tweaks.

**Review by PhotoNomad0 — APPROVED** [review]
LGTM

**Pull request merged**